### ISSUE TITLE
new-cap triggers on Intl.DateTimeFormat

### ISSUE THREAD

**Issue by LandonSchropp** [eslint/eslint#14970]
**Tell us about your environment**

* **ESLint Version:** v7.32.0
* **Node Version:** v14.16.0
* **npm Version:** 6.14.11
* **Operating System:** macOS

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?** @babel/eslint-parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: "@babel/eslint-parser",
  extends: [ "@landonschropp", "@landonschropp/eslint-config-react" ],

  env: {
    es6: true,
    browser: true,
    jest: true
  },

  rules: {

    // TODO: Move this rule to the parent repo.
    "react/jsx-indent-props": [ "warn", { indentMode: 2, ignoreTernaryOperator: true } ],
    "react/sort-comp": "off",

    // TODO: Make sure the  no-global-assign and no-shadow-restricted-names rules are set at the top
    // level so undefined can't be set.
    "no-undefined": "off"
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
Intl.DateTimeFormat().resolvedOptions().timeZone
```

```bash
eslint .
```

**What did you expect to happen?**

This is a built-in API, so it shouldn't produce an error.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

```
52:22  warning  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
```

**Steps to reproduce this issue:**

1. Enable the `no-cap` rule.
2. Add `Intl.DateTimeFormat().resolvedOptions().timeZone` to your code.

**Are you willing to submit a pull request to fix this bug?** Not right now.

**Comment by ljharb** [comment]
`Intl.DateTimeFormat` is a constructor, and should be called with `new`: https://tc39.es/ecma402/#sec-intl-datetimeformat-constructor

Just like a bunch of language builtins, `new` may be optional, but you should still be using it.

**Comment by brettz9** [comment]
Yes, I agree this should be closed as this built-in is indeed a constructor, and if people are enforcing this practice, they may want to require `new` for it. User-land constructors may similarly allow dropping `new` but the rule is meant to enforce this as a practice which makes it visibly distinguishable when a constructor is used.

**Comment by LandonSchropp** [comment]
Ah, gotcha. I didn't realize it was a constructor since many of the examples I looked at weren't calling it with `new`. I'll fix my code. Thanks!
